src/hotspot/share/runtime/disableStackTracingMark.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "runtime/disableStackTracingMark.hpp"
+#include "runtime/javaThread.hpp"
+
+DEBUG_ONLY(THREAD_LOCAL bool DisableStackTracingMark::_is_active = false;)
+
+DisableStackTracingMark::DisableStackTracingMark(JavaThread* jt)
+  : _jt(nullptr),
+    _sp(nullptr) {
+  if (jt == Thread::current()) {
```

Comment:
Just a drive-by comment, but what is the case for `_jt` not being the current thread?

Reply:
If one thread is processing oops of a different thread, for example when you perform a handshake with a target thread and its oops need to be sane, then the jt is the target thread, which is not the current one. But the stack sampling is based on the current thread always, so if the passed in thread isn't the current thread, there is nothing to do. Hmm - I should add a comment shouldn't I? :-)